src/python/WMComponent/AgentStatusWatcher/AgentStatusPoller.py

```diff
@@ -255,29 +260,41 @@ def collectWMBSInfo(self):
 
         return results
 
-    def checkProxyLifetime(self, agInfo):
+    def checkCredLifetime(self, agInfo, credType):
         """
-        Check the proxy lifetime (usually X509_USER_CERT) and raise either
-        a warning or an error if the proxy validity is about to expire.
+        Check the credential lifetime. Usually X509_USER_PROXY or X509_USER_CERT
+        and raise either a warning or an error if the proxy validity is about to expire.
         :param agInfo: dictionary with plenty of agent monitoring information in place.
+        :param credType: credential type, can be: "proxy" or "certificate"
         :return: same dictionary object plus additional keys/values if needed.
         """
-        secsLeft = self.proxy.getTimeLeft(proxy=self.proxyFile)
-        logging.debug("Proxy '%s' lifetime is %d secs", self.proxyFile, secsLeft)
+        if credType == "proxy":
+            credFile = self.proxyFile
+            secsLeft = self.proxy.getTimeLeft(proxy=credFile)
+        elif credType == "certificate":
+            credFile = self.userCertFile
+            secsLeft = self.proxy.getTimeLeft(proxy=credFile, checkVomsLife=False)
+        else:
+            logging.error("Unknown credential type. Available options are: [proxy, certificate]")
+            return
+
+        logging.debug("%s '%s' lifetime is %d seconds", credType, credFile, secsLeft)
+
+        daysLeft = secsLeft / (60. * 60 * 24)
 
-        if secsLeft <= 86400 * 3:  # 3 days
-            proxyWarning = True
+        if daysLeft <= self.credThresholds[credType]['error']:
+            credWarning = True
             agInfo['status'] = "error"
-        elif secsLeft <= 86400 * 5:  # 5 days
-            proxyWarning = True
+        elif daysLeft <= self.credThresholds[credType]['warning']:
+            credWarning = True
             if agInfo['status'] == "ok":
                 agInfo['status'] = "warning"
         else:
-            proxyWarning = False
+            credWarning = False
 
-        if proxyWarning:
-            warnMsg = "Agent proxy '%s' must be renewed ASAP. " % self.proxyFile
+        if credWarning:
+            warnMsg = "Agent %s '%s' must be renewed ASAP." % (credType, credFile)
             warnMsg += "Its time left is: %.2f hours." % (secsLeft / 3600.)
-            agInfo['proxy_warning'] = warnMsg
+            agInfo['%s_warning' % credType] = warnMsg
```

Comment:
This key has to remain `proxy_warning` even if the credential type is a certificate, otherwise this alert won't show up in wmstats (there are further javascript code handling it under the couchapps)

Reply:
Does `agInfo['proxy_warning']` always exist in the dictionary, or do I need to check for it? E.g.:
```
           if agInfo.get('proxy_warning', None):
                agInfo['proxy_warning'] += warnMsg
            else:
                agInfo['proxy_warning'] = warnMsg
```

Reply:
Done.